### PR TITLE
Add juno support

### PR DESCRIPTION
**Pull request by koenkooi**
Meta-96boards is not the right place, but we agreed to put it here till we find a better place.

### CODE REVIEW

File: conf/machine/juno.conf [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#@TYPE: Machine
+#@NAME: Juno vexpress
+#@DESCRIPTION: generic machine to be used by linaro-media-create
```
Comment (fboudra) [inline conf/machine/juno.conf]: Juno ARM Development Platform

File: recipes-bsp/uefi/edk2_git.bb [hunk 2]
```diff
@@ -60,6 +60,13 @@ do_compile() {
 
 do_deploy() {
     # Placeholder to be implemented in machine specific recipe
+
+    if [ -e ${EDK2_DIR}/atf/build/${UEFIMACHINE}/release/bl1.bin ] ; then
```
Comment (fboudra) [inline recipes-bsp/uefi/edk2_git.bb]: why do we need bl1.bin? we don't need to deploy it, it's only used to build fip.bin.

File: conf/machine/juno.conf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            xf86-input-keyboard"
+
+MACHINE_FEATURES = "usbhost usbgadget alsa screen wifi bluetooth efi optee"
+
+PREFERRED_PROVIDER_virtual/kernel ?= "linux-armlt"
+
+KERNEL_IMAGETYPE = "Image"
+KERNEL_DEVICETREE = "arm/juno.dtb arm/juno-r1.dtb arm/juno-r2.dtb"
+
+OPTEEMACHINE = "vexpress-juno"
```
Comment (fboudra) [inline conf/machine/juno.conf]: why do we use OPTEEMACHINE? For HiKey, we use UEFIMACHINE.

It seems we introduce new variables and we don't use them later on (or at least I don't see where we use them).

File: recipes-bsp/uefi/edk2_git.bb [hunk 4]
```diff
@@ -60,6 +60,13 @@ do_compile() {
 
 do_deploy() {
     # Placeholder to be implemented in machine specific recipe
+
+    if [ -e ${EDK2_DIR}/atf/build/${UEFIMACHINE}/release/bl1.bin ] ; then 
+    	install -D -p -m0644 ${EDK2_DIR}/atf/build/${UEFIMACHINE}/release/bl1.bin ${DEPLOYDIR}/bl1.bin
+    fi
+    if [ -e ${EDK2_DIR}/atf/build/${UEFIMACHINE}/release/fip.bin ] ; then
```
Comment (fboudra) [inline recipes-bsp/uefi/edk2_git.bb]: If we install fip.bin in the generic edk2 recipe, we should clean up hikey specific recipe and remove the related install line.

**Review by fboudra — CHANGES_REQUESTED** [review]

File: conf/machine/juno.conf (hunk 1 above)
Comment (koenkooi) [reply conf/machine/juno.conf]: Fixed

File: recipes-bsp/uefi/edk2_git.bb (hunk 2 above)
Comment (koenkooi) [reply recipes-bsp/uefi/edk2_git.bb]: removed

**Pull request merged**